src/tmt_web/utils/git_handler.py

```diff
@@ -73,14 +74,15 @@ def clone_repository(url: str, logger: Logger, ref: str | None = None) -> None:
 
 def get_path_to_repository(url: str) -> Path:
     """
-    Returns the path to the cloned repository from the given URL.
+    Returns the path to the cloned repository from the given URL. The repository url is hashed to
+    avoid repository name collisions.
 
     :param url: URL to the repository
     :return: Path to the cloned repository
     """
-    repo_name = url.rstrip("/").rsplit("/", 1)[-1]
+    clone_dir_name = hashlib.md5(url.encode()).hexdigest()
     root_dir = Path(__file__).resolve().parents[2]  # going up from tmt_web/utils/git_handler.py
-    return root_dir / os.getenv("CLONE_DIR_PATH", "./.repos/") / repo_name
+    return root_dir / os.getenv("CLONE_DIR_PATH", "./.repos/") / clone_dir_name
```

Comment:
I think we should keep the `repo_name` in the folder filename just for the humans (e.g. `<hash>-<repo-name>`). This would be useful if someone wants to remove specific repositories from the repository cache.

Just a thought - if you do not like long filenames, the hash digest can be shortened to the last 10-12 characters. The uniqueness should be still **very** sufficient for our use case.

Reply:
Keeping the repo name as a prefix sounds good to me.

Reply:
> Just a thought - if you do not like long filenames, the hash digest can be shortened to the last 10-12 characters.

Or you can make the hash an actual subdirectory of the repo name. The hierarchy is already encoded in the name with `<hash>-<repo-name>`, why not take advantage of filesystem functionality and transform this encoding into an actual filesystem hierarchy? :)

Reply:
Or we could just skip the hashing entirely and clone into directory named exactly after `url`, that'd make it the clearest. It could work imo, but it think it's safer to get rid of special characters from the `url` by hashing the string.

I'm fine with changing it to `<repo-name>-<hash>` though. :)

Reply:
> I like the `<repo-name>/<hash>` approach, +1 for that.

Yeah, me too.